src/Grid/EventButton.vala

```diff
@@ -49,14 +50,31 @@ public class Maya.View.EventButton : Gtk.Revealer {
         event_box.add (internal_grid);
         event_box.button_press_event.connect ((event) => {
             if (event.type == Gdk.EventType.2BUTTON_PRESS && event.button == Gdk.BUTTON_PRIMARY) {
-                E.Source src = comp.get_data ("source");
-                if (src.writable == true && Model.CalendarModel.get_default ().calclient_is_readonly (src) == false) {
-                    edition_request ();
-                    return true;
+                edit_event ();
+            } else if (event.type == Gdk.EventType.BUTTON_PRESS && event.button == Gdk.BUTTON_SECONDARY) {
+                if (menu == null) {
```

Comment:
It is unnecessary to test for null in creation method (and may not give intended result).  Menu should always be created as it is an instance variable.

Reply:
I was following the existing code style, I have now changed this.